python/lsst/pipe/analysis/coaddAnalysis.py

```diff
@@ -1712,24 +1738,22 @@ def readCatalogs(self, patchRefList, dataset, repoInfo):
         for patchRef in patchRefList:
             if not patchRef.datasetExists(dataset):
                 continue
-            else:
-                patchCat = patchRef.get(dataset, immediate=True, flags=afwTable.SOURCE_IO_NO_FOOTPRINTS)
-                catList.append(patchCat)
-                fname = repoInfo.butler.getUri("deepCoadd_calexp", patchRef.dataId)
-                reader = afwImage.ExposureFitsReader(fname)
-                wcs = reader.readWcs()
-                mask = reader.readMask()
+            patchCat = patchRef.get(dataset, immediate=True, flags=afwTable.SOURCE_IO_NO_FOOTPRINTS)
+            catList.append(patchCat)
+            fname = repoInfo.butler.getUri("deepCoadd_calexp", patchRef.dataId)
+            reader = afwImage.ExposureFitsReader(fname)
+            wcs = reader.readWcs()
+            mask = reader.readMask()
+            maskBad = mask.array & 2**mask.getMaskPlaneDict()["BAD"]
+            maskNoData = mask.array & 2**mask.getMaskPlaneDict()["NO_DATA"]
+            maskedPixels = maskBad + maskNoData
+            numGoodPix = np.count_nonzero(maskedPixels == 0)
```

Comment:
What happens now if the dataset doesn't exist for a given patch? Does it crash? I don't know if that is what we want or not, it is good to know if data you expect to exist doesn't but also it can be useful for it to just get on and process everything that it can.